test/bin/generate_common_versions.py

```diff
@@ -0,0 +1,253 @@
+        return f'"{rc}"'
+
+    ec = get_candidate_repo(minor, dev_preview=True)
+    if repo_exists(ec) and provides_pkg(ec, "microshift"):
+        return f'"{ec}"'
+
+    return '""'
+
+
+def get_release_version(repo):
+    """
+    :param repo: the name or the URL of the repo, enclosed in double quotes
+
+    ``get_release_version`` returns a string to be used as a bash variable,
+    with a call to the right function depending on whether the `repo` param
+    is a URL or not. if neither, returns empty double quotes.
+    """
+    if repo.startswith("\"rhocp"):
+        return f'"$(get_vrel_from_rhsm {repo})"'
+    elif repo.startswith("\"https"):
+        return f'"$(get_vrel_from_beta {repo})"'
+    else:
+        return '""'
+
+
+minor_version = MINOR
+previous_minor_version = minor_version - 1
+yminus2_minor_version = minor_version - 2
+fake_next_minor_version = minor_version + 1
+
+# The current release repository comes from the 'rhocp' stream for release
+# branches, or the OpenShift mirror if only a RC or EC is available. It can
+# be empty, if no candidate for the current minor has been built yet.
+current_release_repo = get_release_repo(minor_version)
+current_release_version = get_release_version(current_release_repo)
+
+# The previous release repository value should either point to the OpenShift
+# mirror URL or the 'rhocp' repository name.
+previous_release_repo = get_release_repo(previous_minor_version)
+previous_release_version = get_release_version(previous_release_repo)
+
+# The y-2 release repository value should either point to the OpenShift
+# mirror URL or the 'rhocp' repository name. It should always come from
+# the 'rhocp' stream.
+yminus2_release_repo = get_release_repo(yminus2_minor_version)
+yminus2_release_version = get_release_version(yminus2_release_repo)
+
+# The 'rhocp_minor_y' variable should be the minor version number, if the
+# current release is available through the 'rhocp' stream, otherwise empty.
+rhocp_minor_y = minor_version if get_sub_repo(minor_version) is not None else '""'
+
+# The beta repository, containing dependencies, should point to the
+# OpenShift mirror URL. If the repository for current minor is not
+# available yet, it should point to an older release.
+rhocp_minor_y_beta = get_beta_repo(minor_version)
+
+# The 'rhocp_minor_y1' should always be the y-1 minor version number.
+rhocp_minor_y1 = previous_minor_version
+rhocp_minor_y1_beta = get_beta_repo(previous_minor_version)
+
+# The 'rhocp_minor_y2' should always be the y-2 minor version number.
+rhocp_minor_y2 = yminus2_minor_version
+
+cncf_sonobuoy_version = "v0.57.3"
+
+print(f"export MINOR_VERSION={minor_version}")
+print(f"export PREVIOUS_MINOR_VERSION={previous_minor_version}")
+print(f"export YMINUS2_MINOR_VERSION={yminus2_minor_version}")
+print(f"export FAKE_NEXT_MINOR_VERSION={fake_next_minor_version}")
+print("")
+
+print(f"CURRENT_RELEASE_REPO={current_release_repo}")
```

Comment:
common_versions.sh must generate arch agnostic url, we cannot commit this because it has hardcoded arch:
```
CURRENT_RELEASE_REPO="https://mirror.openshift.com/pub/openshift-v4/aarch64/microshift/ocp/latest-4.19/el9/os"
```